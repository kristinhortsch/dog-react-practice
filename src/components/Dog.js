import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

function Dog({ dog }) {
  const { name, age, weight } = dog;
  return (
    <Fragment>
      <h2>{name}</h2>
      <p>Age: {age}</p>
      <p>Weight: {weight} lbs</p>
    </Fragment>
  );
}

Dog.propTypes = {
  dog: PropTypes.object.isRequired
};

export default Dog;
