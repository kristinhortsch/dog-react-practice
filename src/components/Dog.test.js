import React from 'React';
import renderer from 'react-test-renderer';
import Dog from './Dog';

describe('Location', () => {
  it('matches a snapshot', () => {
    const dog = {
      name: 'Rover',
      age: 1,
      weight: 12
    };
    const tree = renderer.create(<Dog dog={dog} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
