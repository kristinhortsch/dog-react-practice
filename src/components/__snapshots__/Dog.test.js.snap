// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location matches a snapshot 1`] = `
Array [
  <h2>
    Rover
  </h2>,
  <p>
    Age: 
    1
  </p>,
  <p>
    Weight: 
    12
     lbs
  </p>,
]
`;
